plugins/flytekit-omegaconf/flytekitplugins/omegaconf/dictconfig_transformer.py

```diff
@@ -0,0 +1,178 @@
+                        f"transformers:\n{transformation_logs}"
+                    )
+
+        wrapper = Struct()
+        wrapper.update(
+            flatten_dict.flatten(
+                {
+                    "types": type_map,
+                    "values": value_map,
+                    "base_dataclass": f"{base_config.__module__}.{base_config.__name__}",
+                },
+                reducer="dot",
+                keep_empty_types=(dict,),
+            )
+        )
+        return Literal(scalar=Scalar(generic=wrapper))
+
+    def to_python_value(self, ctx: FlyteContext, lv: Literal, expected_python_type: Type[DictConfig]) -> DictConfig:
+        """Re-hydrate the custom object from Flyte Literal value."""
+        if lv and lv.scalar is not None:
+            nested_dict = flatten_dict.unflatten(MessageToDict(lv.scalar.generic), splitter="dot")
+            cfg_dict = {}
+            for key, type_desc in nested_dict["types"].items():
+                if type_desc == "MISSING":
+                    cfg_dict[key] = omegaconf.MISSING
+                else:
+                    if re.match(r".+\[.*]", type_desc):
+                        origin_module, origin_type = type_desc.split("[")[0].rsplit(".", 1)
+                        origin = importlib.import_module(origin_module).__getattribute__(origin_type)
+                        sub_types = type_desc.split("[")[1][:-1].split(", ")
+                        for i, t in enumerate(sub_types):
+                            if t != "NoneType":
+                                module_name, class_name = t.rsplit(".", 1)
+                                sub_types[i] = importlib.import_module(module_name).__getattribute__(class_name)
+                            else:
+                                sub_types[i] = type(None)
+                        node_type = origin[tuple(sub_types)]
+                    else:
+                        module_name, class_name = type_desc.rsplit(".", 1)
+                        node_type = importlib.import_module(module_name).__getattribute__(class_name)
+
+                    transformation_logs = ""
+                    transformer = TypeEngine.get_transformer(node_type)
+                    value_literal = Literal.from_flyte_idl(ParseDict(nested_dict["values"][key], PB_Literal()))
+                    cfg_dict[key] = transformer.to_python_value(ctx, value_literal, node_type)
+
+                    if key not in cfg_dict.keys():
+                        raise ValueError(
+                            f"Could not identify matching transformer for object {nested_dict['values'][key]} of "
+                            f"proposed type {node_type}. This may either indicate that no such transformer "
+                            "exists or the appropriate transformer cannot deserialise this object. Attempted the "
+                            f"following transformers:\n{transformation_logs}"
```

Comment:
@mg515 I need some clarification here, sorry :)

Is it still true in the current version of the PR that the dict config type transformer tries different type transformers for serializing the dict config nodes? When searching for `transformation_logs`, I can only see the variable being set to an empty string and being used in the `VallueError` messages. Can you please point me to where we iterate over different transformers and where `transformation_logs` is being appended to?

Reply:
bump.

Reply:
Yeah, the warning is obsolete with the removal of iterating through type transformers, my bad. Addressed now in https://github.com/flyteorg/flytekit/pull/2299/commits/709d4a4364e94f2e531f533c076b564dc3a26d38